Update Owncast developer documentation to mention Windows #2864

dev265545 opened the pull request:
- Added a note for windows users in the documentation.
- Adding owncast_for_windows file - a guide to install and run Owncast in Windows using WSL2
-  Corrected some grammatical mistakes

dev265545 committed:
Update README.md

dev265545 committed:
Update README.md

dev265545 committed:
Update README.md

dev265545 committed:
Update README.md

dev265545 committed:
Merge branch 'develop' of https://github.com/dev265545/owncast into develop

dev265545 committed:
Create owncast_for_windows.md

dev265545 committed:
Update README.md

dev265545 committed:
Update README.md

dev265545 committed:
Update owncast_for_windows.md

gabek commented:
Thanks so much for working on this!

>This document list out the steps in detail to install and run Owncast in Windows using Windows Subsystem for Linux

This mentions it's for _running_ Owncast, but really it's the instructions for _building_ Owncast. I think we should be clear if the instructions are for people trying to install and run Owncast as a user, or build and work on Owncast as a developer. I think these instructions are specifically for building and working on Owncast since discuss building the source.

Would you mind adding that clarification?

If you want to have a section for _installing_ Owncast under Windows that'd be great too, however!

dev265545 commented:
Yes, I will make the changes and add the steps for installing Owncast as a user.

dev265545 committed:
Update README.md

dev265545 committed:
Update owncast_for_windows.md

More clarity about different use cases
Added steps for installing Owncast as a user in Windows.

gabek commented on owncast_for_windows.md:
```diff
@@ -52,20 +53,40 @@ Here is the list for all the prerequisites required ->
 1. Point your [broadcasting software](https://owncast.online/docs/broadcasting/) at your new server and start streaming.
 
-### Step 4: Installing Owncast Frontend
+### Step 6: Installing Owncast Frontend
```
Sorry to be picky, but these steps as stated are a bit confusing. It says you have to 1) Install Owncast 2) Setup the backend 3) Setup the frontend.

Maybe _using_ owncast and _developing_ owncast should be split up so it doesn't look like you have to install clone the repo and run the backend and frontend in order to run the software.

dev265545 committed:
Update owncast_for_windows.md

Changing the format a little bit

gabek commented on owncast_for_windows.md:
```diff
@@ -0,0 +1,139 @@
+
+<hr>
+
+## Now for users who just want to use Owncast
```
Since more people want to simply run Owncast than to develop Owncast the "use" should probably be before the "develop" part, because most people don't need to know about compilers, nodejs, etc.

gabek commented on owncast_for_windows.md:
```diff
@@ -0,0 +1,139 @@
+##  Next steps are for developers and users looking to build and install Owncast.
+These steps are for building and install Owncast on Windows for development and contributing to this project.
+
+### Installing Owncast Backend
```
Since it's not really "installing" maybe saying "Running Owncast backend from source" is another way to say it.

gabek commented on owncast_for_windows.md:
```diff
@@ -0,0 +1,139 @@
+1. Visit `http://yourserver:8080` to access the web interface or `http://yourserver:8080/admin` to access the admin.
+1. Point your [broadcasting software](https://owncast.online/docs/broadcasting/) at your new server and start streaming.
+
+### Installing Owncast Frontend
```
Same here, it's not really installing anything, just running the source.

dev265545 committed:
Update owncast_for_windows.md

gabek commented:
Thanks for all the updates! :1st_place_medal:
